### ISSUE TITLE
Pacemaker restart failed

### ISSUE THREAD

**Issue by simonzg** [meterio/meter-pov#20]
Reported by: DavidRise

David Rise, [2023/1/11 15:28]
In October, this was almost non-existent. It was very rare.

David Rise, [2023/1/11 15:31]
The second node is the only one that I didn't reload. Yes, she is now in jail. This problem is solved by rebooting the periodic. As 40-60 fines are collected, I restart the server and they subside.

David Rise, [2023/1/11 15:32]
The longer the server works, the more often the fines, as I understand it. This was not the case 2 months ago.

David Rise, [2023/1/11 15:34]
I have rebooted the server. We will watch. Now fines should not be quickly recruited.

**Comment by simonzg** [comment]
Corresponding Logs

```
t=2023-01-11T21:40:22+0000 lvl=warn msg="pre-commit stage is empty" pkg=reactor
t=2023-01-11T21:40:22+0000 lvl=info msg="Validated block proposal" pkg=pacemaker height=32320023 id=0x01ed2a1762e36bcd71b56a52f736a5977aab8582727b0d07b827ad4f68c9676e
t=2023-01-11T21:40:22+0000 lvl=info msg="Send>> [PMVote Height:32320023 Round:299]" pkg=pacemaker to=CLK_2(52.12.91.2) msgHash=47291dee
t=2023-01-11T21:40:22+0000 lvl=info msg="Update bestQC from bestBlock descendant" pkg=chain from="QC(H:32320027, R:303, Epoch:20942)" to="QC(H:32320028, R:304, Epoch:20942)"
block already in chain:  32320020 0x01ed2a14e79dcf6216c7bae3dfbae8c6870c03e539a94ab01b7f79c2d90a6d1d
t=2023-01-11T21:40:22+0000 lvl=info msg="Found pending message" pkg=pacemaker height=32320023
t=2023-01-11T21:40:22+0000 lvl=info msg="Stop round timer" pkg=pacemaker round=299
t=2023-01-11T21:40:22+0000 lvl=info msg="Start round timer" pkg=pacemaker round=300 counter=0 power=0 interval=16
t=2023-01-11T21:40:22+0000 lvl=info msg="update current round" pkg=pacemaker to=300 reason=RegularProposal
t=2023-01-11T21:40:22+0000 lvl=warn msg="pre-commit stage is empty" pkg=reactor
t=2023-01-11T21:40:22+0000 lvl=info msg="Validated block proposal" pkg=pacemaker height=32320024 id=0x01ed2a181bfaeade055183ba361e403567694020cd9c48f5d915a31b7cc33879
t=2023-01-11T21:40:22+0000 lvl=info msg="Send>> [PMVote Height:32320024 Round:300]" pkg=pacemaker to=Kevvonode(3.9.168.58) msgHash=47aec298
t=2023-01-11T21:40:22+0000 lvl=info msg="Update bestQC from bestBlock descendant" pkg=chain from="QC(H:32320028, R:304, Epoch:20942)" to="QC(H:32320029, R:305, Epoch:20942)"
block already in chain:  32320021 0x01ed2a1522de56670e312ca6d5e4427ede816d08fd0cfc6f36e0618de8e40f72
t=2023-01-11T21:40:22+0000 lvl=info msg="Stop round timer" pkg=pacemaker round=300
t=2023-01-11T21:40:22+0000 lvl=info msg="Start round timer" pkg=pacemaker round=301 counter=0 power=0 interval=16
t=2023-01-11T21:40:22+0000 lvl=info msg="update current round" pkg=pacemaker to=301 reason=RegularProposal
t=2023-01-11T21:40:22+0000 lvl=warn msg="pre-commit stage is empty" pkg=reactor
t=2023-01-11T21:40:22+0000 lvl=info msg="Validated block proposal" pkg=pacemaker height=32320025 id=0x01ed2a19c4cf0daedb73ba8300f3881d9cebd4112b3a71d733cfd0be20c44635
t=2023-01-11T21:40:22+0000 lvl=info msg="Send>> [PMVote Height:32320025 Round:301]" pkg=pacemaker to=RedBull-US(45.148.29.35) msgHash=63ca70cf
t=2023-01-11T21:40:22+0000 lvl=info msg="Update bestQC from bestBlock descendant" pkg=chain from="QC(H:32320029, R:305, Epoch:20942)" to="QC(H:32320030, R:306, Epoch:20942)"
block already in chain:  32320022 0x01ed2a16025b890069c364c00dac8f9cbedcff5f14da509236528855ccb666c0
t=2023-01-11T21:40:22+0000 lvl=info msg="Send>> [PMQueryProposal FromHeight:32320025 ToHeight:32320026 QueryRound:302]" pkg=pacemaker to=DFL1(34.223.162.7) msgHash=ec5cc6fb
Error happend during OnReceiveProposal: parent block is missing
t=2023-01-11T21:40:22+0000 lvl=warn msg="Pacemaker restart requested" pkg=consensus qcHigh="pmQC{QC:(H:32320024,R:300), qcNode:(H:32320024,R:300)}" qcHigh.QCNode=32320024 err="parent block is missing"
t=2023-01-11T21:40:22+0000 lvl=warn msg="Process PMProposal failed" pkg=pacemaker err="parent block is missing"
t=2023-01-11T21:40:22+0000 lvl=info msg="Start to execute cmd" pkg=pacemaker cmd=Restart
t=2023-01-11T21:40:22+0000 lvl=info msg="Recv [PMProposal 19f4cbf7 (H:32320067,R:343), Parent:(H:32320066,R:342), Proposed:{mBlock(32320067), ID:0x01ed2a43�~@�acb1af80, QC:(H:32320066,R:342)}]" pkg=pacemaker peer=BIWMeterNode3U ip=52.62.70.249 msgCh=40/1024
t=2023-01-11T21:40:22+0000 lvl=info msg="Relay>> [PMProposal (H:32320067,R:343), Parent:(H:32320066,R:342), Proposed:{mBlock(32320067), ID:0x01ed2a43�~@�acb1af80, QC:(H:32320066,R:342)}]" pkg=pacemaker to="ZEUS CryptoNode(49.12.240.199), Alo's Meter Wildlife Fund(74.207.243.213), HashFuture1(207.180.193.129), JamesMeterian_ITALY89�~_~G��~_~G�(23.88.103.223)" height=32320067 round=343 msgHash=19f4cbf7
t=2023-01-11T21:40:25+0000 lvl=info msg="Recv [PMProposal a5978209 (H:32320068,R:344), Parent:(H:32320067,R:343), Proposed:{mBlock(32320068), ID:0x01ed2a44�~@�c97cbf3a, QC:(H:32320067,R:343)}]" pkg=pacemaker peer="Considerate Culture " ip=50.19.206.74 msgCh=41/1024
```

**Comment by simonzg** [comment]
From log it seems like it's caused by a deadlock on rw lock, as fixed in bd3a0d49918beb8970c8f804a76fa0fc3542d964
